File: Python/0048_rotate_image/0048_rotate_image.py
class Solution:
    def rotate(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: void Do not return anything, modify matrix in-place instead.
        """
        for i in range(0, len(matrix) // 2):
            for j in range(0, len(matrix)):
                matrix[len(matrix) - 1 - i][j], matrix[i][j] = matrix[i][j], matrix[len(matrix) - 1 - i][j]
        for i in range(0, len(matrix)):
            for j in range(i + 1, len(matrix)):
                matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]

File: Java/0048_rotate_image/0048_rotate_image.java
class Solution {
    public void rotate(int[][] matrix) {
        reverse(matrix);
        for (int i = 0; i < matrix.length; i++) {
            for (int j = i + 1; j < matrix.length; j++) {
                int temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp;
            }
        }
    }
    
    public void reverse(int[][] matrix) {
        for (int i = 0; i < matrix.length / 2; i++) {
            for (int j = 0; j < matrix.length; j++) {
                int temp = matrix[matrix.length - 1 - i][j];
                matrix[matrix.length - 1 - i][j] = matrix[i][j];
                matrix[i][j] = temp;
            }
        }
    }
}